services/src/integ-test/java/uk/gov/moj/sdt/services/messaging/MessageWriterIntTest.java

```diff
@@ -109,20 +89,7 @@ public void testQueueMessage() throws JMSException, InterruptedException {
         message2.setSdtRequestReference(strMessage2);
         messageWriter.queueMessage(message2, "TEST1", false);
 
-        // Read the two messages and ensure they are read back in the same order.
-        ServiceBusReceivedMessage message = messageReceiver.receiveMessage("Test1Queue");
-        if (message == null) {
-            Assert.fail("Test failed because JMS receive timed out.");
-        }
-
-        ISdtMessage sdtMessage = message.getBody().toObject(new TypeReference<SdtMessage>() {});
-        LOGGER.debug("Message Receieved 1 - " + sdtMessage.toString());
-        Assert.assertTrue(sdtMessage.getSdtRequestReference().equals(strMessage1));
-
-        message = messageReceiver.receiveMessage("Test1Queue");
-        sdtMessage = message.getBody().toObject(new TypeReference<SdtMessage>() {});
-        LOGGER.debug("Message Receieved2 - " + sdtMessage.toString());
-        Assert.assertTrue(sdtMessage.getSdtRequestReference().equals(strMessage2));
+        readMessageFromQueue(3);
```

Comment:
The original test checked the content of the messages to ensure they were retrieved in the expected order - now it only checks the number of messages.  Test should have the functionality it originally had.

Reply:
need to discuss about this